### DISCUSSION TITLE
Realistically, how large can Concourse scale?

### DISCUSSION THREAD

**Discussion by jriddy** [concourse/concourse#8764] [General]
I'm looking at the feasibility of migrating from a Jenkins instance that manages about 250 agents in the cloud now to Concourse.  One of the reasons we're looking at migrating is because we'd like to scale up the number or workers dramatically. Many of our workloads tend to require all of a machine's resources, so I'm assuming a 1:1 task:worker ratio for simplicity.

How many worker nodes have we seen Concourse scale to in the wild? What factors does this depend on? Is  there a different recommendation of worker nodes to web nodes if you have your workers set low low task limits?

I'm guessing the overall scaling relies primarily on the setup of the PG database powering it all, but I want to know if there are other scaling constratints, or even just "how big have we seen this get in the wild?"

**Comment by evanchaoli** [comment]
@jriddy Our biggest production cluster has 7 ATCs and 220 workers. The cluster hosts 1000+ teams and 13000+ pipelines, and it runs stable. The database size is now 250GB. If you want to deploy a large cluster like ours, you must use a version of 7.8+.

**Reply by jriddy** [reply]
Neat! Do you have any kind of autoscaling on for your workers?

Seems like you have a lot more pipelines than we would, which I imagine is where the data comes from. But fewer workers.  Guess you can scale ATC and TSA independently

**Comment by evanchaoli** [comment]
We don't auto-scale workers. We have a self-developed worker-manager that monitors worker status, auto retire unhealthy workers. And we semi-auto worker-scale, we manually set worker number, then worker-manager will ensure there are configured number of workers.

ATC and TSA is integrated together as a `concourse web`, and we call it ATC. So we don't treat ATC and TSA independently.
